client/block.c

```diff
@@ -1607,7 +1607,7 @@ int xdag_print_block_info(xdag_hash_t hash, FILE *out)
 	fprintf(out, "                               block as transaction: details\n");
 	fprintf(out, " direction  address                                    amount\n");
 	fprintf(out, "-------------------------------------------------------------------------------------------\n");
-	if(bi->ref) {
+	if((bi->flags & BI_REF) && bi->ref != NULL) {
```

Comment:
@sgaragagghu  Here is my comments, if bi is still not referred by main block, it should be queryable, and we should know what is the block for fee rather than set the fee block as AAAAA. Well, with Daniel's fix of extra blocks, it may make sense, but have to see.

Reply:
reference block will be setted when BI_REF is setted, there is no fee block until the block is not BI_REF. Before extra block, ref was NULL until BI_REF was setted, but now it is different, bi->ref is != NULL before BI_REF is setted, so you are deferencing bi->ref as if it was block_internal but it is a block_orphan, you will see a strange fee address if you try without this fix.

Reply:
Okay, see the point. So it's block_orphan related. Let me check.